pkg/resources/materialized_view.go

```diff
@@ -0,0 +1,299 @@
+
+	extractor := snowflake.NewViewSelectStatementExtractor(v.Text.String)
+	substringOfQuery, err := extractor.Extract()
+	if err != nil {
+		return err
+	}
+
+	err = d.Set("statement", substringOfQuery)
+	if err != nil {
+		return err
+	}
+
+	return d.Set("database", v.DatabaseName.String)
+}
+
+// UpdateMaterializedView implements schema.UpdateFunc
+func UpdateMaterializedView(d *schema.ResourceData, meta interface{}) error {
+	dbName, schema, view, err := splitViewID(d.Id())
+	if err != nil {
+		return err
+	}
+
+	builder := snowflake.MaterializedView(view).WithDB(dbName).WithSchema(schema)
+
+	db := meta.(*sql.DB)
+	if d.HasChange("name") {
+		name := d.Get("name")
+
+		q := builder.Rename(name.(string))
+		err := snowflake.Exec(db, q)
+		if err != nil {
+			return errors.Wrapf(err, "error renaming view %v", d.Id())
+		}
+
+		d.SetId(fmt.Sprintf("%v|%v|%v", dbName, schema, name.(string)))
+	}
+
+	if d.HasChange("comment") {
+		comment := d.Get("comment")
+
+		if c := comment.(string); c == "" {
+			q := builder.RemoveComment()
+			err := snowflake.Exec(db, q)
+			if err != nil {
+				return errors.Wrapf(err, "error unsetting comment for view %v", d.Id())
+			}
+		} else {
+			q := builder.ChangeComment(c)
+			err := snowflake.Exec(db, q)
+			if err != nil {
+				return errors.Wrapf(err, "error updating comment for view %v", d.Id())
+			}
+		}
+	}
+	if d.HasChange("is_secure") {
+		secure := d.Get("is_secure")
+
+		if secure.(bool) {
+			q := builder.Secure()
+			err := snowflake.Exec(db, q)
+			if err != nil {
+				return errors.Wrapf(err, "error setting secure for view %v", d.Id())
+			}
+		} else {
+			q := builder.Unsecure()
+			err := snowflake.Exec(db, q)
+			if err != nil {
+				return errors.Wrapf(err, "error unsetting secure for materialized view %v", d.Id())
+			}
+		}
+	}
+
+	if d.HasChange("cluster_by") {
+		clusterBy := d.Get("cluster_by").([]interface{})
+		if len(clusterBy) == 0 {
+			q := builder.RemoveClusterBy()
+			err := snowflake.Exec(db, q)
+			if err != nil {
+				return errors.Wrapf(err, "error unsetting cluster_by for view %v", d.Id())
+			}
+		} else {
+			q := builder.ChangeClusterBy(clusterBy)
+			err := snowflake.Exec(db, q)
+			if err != nil {
+				return errors.Wrapf(err, "error updating cluster_by for view %v", d.Id())
+			}
+		}
+	}
+
+	return ReadMaterializedView(d, meta)
+}
+
+// DeleteMaterializedView implements schema.DeleteFunc
+func DeleteMaterializedView(d *schema.ResourceData, meta interface{}) error {
+	db := meta.(*sql.DB)
+	dbName, schema, view, err := splitViewID(d.Id())
+	if err != nil {
+		return err
+	}
+
+	q := snowflake.MaterializedView(view).WithDB(dbName).WithSchema(schema).Drop()
+
+	err = snowflake.Exec(db, q)
+	if err != nil {
+		return errors.Wrapf(err, "error deleting materialized view %v", d.Id())
+	}
+
+	d.SetId("")
+
+	return nil
+}
+
+// MaterializedViewExists implements schema.ExistsFunc
```

Comment:
No need for an [ExistsFunc](https://www.terraform.io/docs/extend/guides/v2-upgrade-guide.html#deprecation-of-helper-schema-existsfunc) anymore.